packages/toggle/src/injectToggle.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { InjectComponentReturnType } from '@udecode/plate-common';
+
+import { useToggleControllerStore, useToggleIndex } from './store';
+
+export const injectToggle = (): InjectComponentReturnType => WithToggle;
+
+const WithToggle: InjectComponentReturnType = ({ element, children }) => {
+  // Instead of using both the toggle index and the open ids atoms,
+  //    use a single atom that is subscribed to the sole value relevant for this element: isOpen
+  const [openIds] = useToggleControllerStore().use.openIds();
+  const toggleIndex = useToggleIndex();
+  const enclosedInToggleIds = toggleIndex.get(element.id as string) || [];
+  const isOpen = enclosedInToggleIds.every((id) => openIds.has(id));
```

Comment:
Something like?

```ts
export const isOpenAtom = atom((get) => {
  const openIds = get(toggleControllerStore.atom.openIds)
  const toggleIndex = get(toggleIndexAtom)
  const enclosedInToggleIds = toggleIndex.get(element.id as string) || [];

  return enclosedInToggleIds.every((id) => openIds.has(id))
})
```

Reply:
From this example: https://github.com/udecode/plate/blob/6d027463a88dbaa4f4bd02330a132956f2094597/packages/core/src/stores/plate-controller/plateControllerStore.ts#L34

```ts
export const useIsToggleOpen = (id: string) => {
  const isOpenAtom = useMemo(
      () => atom((get) => {
        const openIds = get(toggleControllerStore.atom.openIds)
        const toggleIndex = get(toggleIndexAtom)
        const enclosedInToggleIds = toggleIndex.get(id) || [];
      
        return enclosedInToggleIds.every((enclosedId) => openIds.has(enclosedId))
      }),
      [element.id],
// ...

  return useToggleControllerStore().get.atom(isOpenAtom)
```

Reply:
No luck. [This part of jotai-x's doc](https://github.com/udecode/jotai-x?tab=readme-ov-file#externally-defined-derived-atoms) says that you can derive state from a store created with `createAtomStore`. In our case the state derives from both `toggleControllerStore` and `plateEditorStore`. So I guess that at least the `useIsToggleOpen` hook should be symmetric in its use of `toggleControllerStore` and `plateEditorStore`.

Using your sample @zbeyens, the toggleIndexAtom stays blocked at an empty map.
Using `return usePlateStore().get.atom(isOpenAtom)` instead of `return useToggleControllerStore().get.atom(isOpenAtom)`, the `openIds` atom stays blocked at an empty array.

I'm so confused 😆

Reply:
I think this is a limitation of Jotai; you can't create a derived atom that depends on multiple stores. I've definitely encountered this problem myself. My advice would be to use a different hook for each store, and put the one that's least likely to change first to reduce unnecessary re-renders.

Reply:
That's a good idea, opening and closing toggles will surely be much less frequent than updating children 👌 

I guess that's what you meant but jotai can definitely derive from multiple atoms, it must be `jotai-x` that stands in the way.

Just tested, it works without and without constant re-renders.

Reply:
Deriving from multiple atoms works fine, but not when those atoms come from different JotaiStores, which is where the limitation comes from.

Reply:
gotcha 👌